packages/gatsby-image/src/index.js

```diff
@@ -395,8 +392,8 @@ class Image extends React.Component {
       className: placeholderClassName,
     }
 
-    if (fluid || fluidImages) {
-      const imageVariants = fluidImages ? groupByMedia(fluidImages) : [fluid]
+    if (fluid.length) {
+      const imageVariants = groupByMedia(fluid)
```

Comment:
I see that you've moved this logic to `convertProps()` but left the `groupByMedia()` method here. My only concern about that is the render methods referencing `imageVariants` will now potentially have a different order of elements than anything referencing fluid/fixed by index directly. Presently, that's only `getImageSrcKey()`. 

I would like to properly address returning the correct key for a variant in future so that `image` will not have the issues it presently will with this feature when a variant differs from properties that `image` is used for instead.

Should `groupByMedia()` be moved to `convertProps()` as well? If there's any concerns for additional overhead, the result could always be memoized.

Reply:
Like earlier, what is the `length` here for? If you want to check this is an array, use `Array.isArray(fluid)`, although it should always be given `convertProps()`?

Reply:
Moved groupByMedia to convertedProps, that's a good suggestion. Even if it might slow down a bit. It's clearer in code. I'll refactor it a little in another PR so negate this effect.